core/src/main/scala/za/co/absa/fadb/statushandling/StatusException.scala

```diff
@@ -20,22 +20,56 @@ import za.co.absa.fadb.exceptions.DBFailException
 
 /**
   * Exception caused by status signaling a failure in DB function execution
-  * @param status     - the status that caused the error
-  * @param statusText - the status text explaining the status code
+  * @param status     - represent the status information returned from the function call
   */
-class StatusException(val status: Int, statusText: String) extends DBFailException(statusText) {
-  def statusText: String = getMessage
+class StatusException(val status:FunctionStatus) extends DBFailException(status.statusText) {
+
+  override def equals(obj: Any): Boolean = {
```

Comment:
why this `equals` is here and not in `DBFailException`?

Reply:
yeah, the implementation would differ, but the whole 'chain' of these particular exception hierarchy would have it

Reply:
OK, will add it. Useful in test case.